numpy/core/src/multiarray/arrayobject.c

```diff
@@ -661,36 +534,42 @@ static PyObject *
 array_repr(PyArrayObject *self)
 {
     PyObject *s, *arglist;
+    PyObject *reprfunc = NULL;
 
-    if (PyArray_ReprFunction == NULL) {
-        s = array_repr_builtin(self, 1);
+    npy_cache_import("numpy.core._internal", "_pyarray_repr_function", &reprfunc);
```

Comment:
So the idea is to (also) cache the variable in python instead of keep it around only inside C I guess? Two thing are not quite clear to me.
1. What happens with NULL? My guess is, it would delete the python side function, and then it errors here, the error is swallowed and the old (working) cached function is used?! This happens when things get set to None (which is translated to NULL) if I see right.
2. Why does this work ;). I am serious. The npy_cache_import should not get updated after the first time?! But the repr function can change later on.

There seems a test for both of it, so I am not too worried it is broken, but still a bit puzzled.